pkg/clusteragent/autoscaling/workload/controller.go

```diff
@@ -463,3 +490,14 @@ func (c *Controller) updateAutoscalerStatusAndUnlock(ctx context.Context, key, n
 	c.store.UnlockSet(key, podAutoscalerInternal, c.ID)
 	return err
 }
+
+func getActiveSource(currentTime time.Time, mainHorizontalScalingValues *model.HorizontalScalingValues) datadoghq.DatadogPodAutoscalerValueSource {
+	if mainHorizontalScalingValues == nil {
```

Comment:
I think this case happen at startup for some time (leader election + RC refresh) and I would not want to trigger local fallback in this case.
It could also happen if it's a freshly created Autoscaler and similarly, we don't want to trigger local fallback in this case.

Also the local fallback can fail. If the data from local fallback is also stale we should probably either:
- Do nothing (keep the current values)
- Use the most recent values

Reply:
Updating this so `getActiveSource` now returns:
* `nil` if `mainHorizontalScalingValues` is `nil` -> doesn't update `scalingValues` (assumes this is from the case where leader election change happened/on startup where we deserialize from `Status()`)
* `Local` if local fallback recommendations are not stale and product values are stale
* `Autoscaling` otherwise